vochain/indexer/indexer.go

```diff
@@ -225,16 +206,38 @@ func (idx *Indexer) Close() error {
 	return nil
 }
 
-// BackupPath restores the database from a backup created via SaveBackup.
+// RestoreBackupFromFile restores the database from a backup file created via SaveBackup.
+// Note that this must be called with ExpectBackupRestore set to true,
+// and before any indexing or queries happen.
+func (idx *Indexer) RestoreBackupFromFile(_ context.Context, path string) error {
+	srcf, err := os.Open(path)
+	if err != nil {
+		return err
+	}
+	defer srcf.Close()
+
+	return idx.RestoreBackup(srcf)
+}
+
+// RestoreBackup restores the database from a backup created via SaveBackup.
 // Note that this must be called with ExpectBackupRestore set to true,
 // and before any indexing or queries happen.
-func (idx *Indexer) RestoreBackup(ctx context.Context, path string) error {
+func (idx *Indexer) RestoreBackup(r io.Reader) error {
 	if idx.readWriteDB != nil {
 		panic("Indexer.RestoreBackup called after the database was initialized")
 	}
-	if err := copyFile(idx.dbPath, path); err != nil {
+
+	log.Debugf("restoring indexer backup into %s", idx.dbPath)
+	file, err := os.Create(idx.dbPath)
+	if err != nil {
+		return err
+	}
+	defer file.Close()
```

Comment:
FYI there's a regression here: you're no longer checkig for the file close error, which copyFile did.

Reply:
sorry about that, reverted back to copyFile